### PR TITLE
Write displacement on .h5 files

### PR DESCRIPTION
**Pull request by ezrayst**
- [x] Add `displacement` to `hdf5.h`
- [x] Write `displacement` to .h5 files in `mesh.tcc`
- [x] Read `displacement` from .h5 files in `mesh.tcc`

### CODE REVIEW

File: include/mesh.tcc [hunk 1]
```diff
@@ -918,51 +927,109 @@ bool mpm::Mesh<Tdim>::write_particles_hdf5(unsigned phase,
...
-      HOFFSET(HDF5Particle, velocity_x), HOFFSET(HDF5Particle, velocity_y),
-      HOFFSET(HDF5Particle, velocity_z), HOFFSET(HDF5Particle, stress_xx),
-      HOFFSET(HDF5Particle, stress_yy),  HOFFSET(HDF5Particle, stress_zz),
-      HOFFSET(HDF5Particle, tau_xy),     HOFFSET(HDF5Particle, tau_yz),
-      HOFFSET(HDF5Particle, tau_xz),     HOFFSET(HDF5Particle, strain_xx),
-      HOFFSET(HDF5Particle, strain_yy),  HOFFSET(HDF5Particle, strain_zz),
-      HOFFSET(HDF5Particle, gamma_xy),   HOFFSET(HDF5Particle, gamma_yz),
-      HOFFSET(HDF5Particle, gamma_xz),   HOFFSET(HDF5Particle, epsilon_v),
-      HOFFSET(HDF5Particle, status),     HOFFSET(HDF5Particle, cell_id),
+      HOFFSET(HDF5Particle, id),
```
Comment (kks32) [inline include/mesh.tcc:934]: Could you reformat to the two columns it was like before?

File: include/mesh.tcc [hunk 2]
```diff
@@ -918,51 +927,109 @@ bool mpm::Mesh<Tdim>::write_particles_hdf5(unsigned phase,
...
 
   // Define particle field information
-  const char* field_names[NFIELDS] = {
-      "id",         "mass",       "volume",     "pressure",  "coord_x",
-      "coord_y",    "coord_z",    "nsize_x",    "nsize_y",   "nsize_z",
-      "velocity_x", "velocity_y", "velocity_z", "stress_xx", "stress_yy",
-      "stress_zz",  "tau_xy",     "tau_yz",     "tau_xz",    "strain_xx",
-      "strain_yy",  "strain_zz",  "gamma_xy",   "gamma_yz",  "gamma_xz",
-      "epsilon_v",  "status",     "cell_id"};
+  const char* field_names[NFIELDS] = {"id",
```
Comment (kks32) [inline include/mesh.tcc:1002]: Could you reformat to as it was like before?

File: include/mesh.tcc [hunk 3]
```diff
@@ -1028,44 +1098,78 @@ bool mpm::Mesh<Tdim>::read_particles_hdf5(unsigned phase,
...
-      sizeof(particle.velocity_x), sizeof(particle.velocity_y),
-      sizeof(particle.velocity_z), sizeof(particle.stress_xx),
-      sizeof(particle.stress_yy),  sizeof(particle.stress_zz),
-      sizeof(particle.tau_xy),     sizeof(particle.tau_yz),
-      sizeof(particle.tau_xz),     sizeof(particle.strain_xx),
-      sizeof(particle.strain_yy),  sizeof(particle.strain_zz),
-      sizeof(particle.gamma_xy),   sizeof(particle.gamma_yz),
-      sizeof(particle.gamma_xz),   sizeof(particle.epsilon_v),
-      sizeof(particle.status),     sizeof(particle.cell_id),
+      sizeof(particle.id),
```
Comment (kks32) [inline include/mesh.tcc:1142]: Could you reformat to as it was like before?

**Review by kks32 — CHANGES_REQUESTED** [review]
Do we need to store displacement for resume function call?

File: include/mesh.tcc (hunk 3 above)
Comment (ezrayst) [reply include/mesh.tcc:1142]: It is done automatically by `clang`. We can revert back maybe by changing from `displacement` to `disp`. Do you want me to do that?

Comment (kks32) [reply include/mesh.tcc:1142]: OK leave it as is

**Review by kks32 — CHANGES_REQUESTED** [review]
You need to set displacement on resume

**Comment by ezrayst** [comment]
Displacement initialization in `particle` and test cases are added

**Review by kks32 — APPROVED** [review]

**Pull request merged**